### ISSUE TITLE
Logger infers Pino as Bunyan due to shared prototype (.child() method)

### ISSUE THREAD

**Issue by teijo** [instana/nodejs-sensor#119]
Hi,

I tried to use [Pino logger](https://github.com/pinojs/pino) with the 1.60.0 version of the package and noticed that `src/logger.js` gets into conflict with it. On the following lines, logger tries to determine whether Bunyan logger or another logger is provided based on existence of `.child()` method.

https://github.com/instana/nodejs-sensor/blob/9e224cbce35837a34b147c5de886c5d54fc4ce56/src/logger.js#L13-L19

However, [Pino also has a that function](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/pino/index.d.ts#L381).

The problem arises when Bunyan is incorrectly inferred, and then follows up trying to call Bunyan-specific `.addStream()`:

https://github.com/instana/nodejs-sensor/blob/9e224cbce35837a34b147c5de886c5d54fc4ce56/src/logger.js#L33-L34

Determining whether the logger is Bunyan would need to be more reliable, or perhaps the additional check for `.addStream()` could be introduced.

Example workaround in the meanwhile (in JavaScript, replace `(logger as any)` with just `logger`):

```ts
const child = logger.child;
(logger as any).child = undefined; // Remove for initialization
instana({ logger }); // or instana.setLogger(logger)
logger.child = child; // Restore
```

**Comment by basti1302** [comment]
Hi,

thanks for the report. I'll take care of it.

Kind regards

  Bastian

**Comment by basti1302** [comment]
A fix has been published with 1.60.1. Thanks again for the detailed report 👍
